frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx

```diff
@@ -137,7 +138,8 @@ class Dashboard extends Component {
       this.throttleParameterWidgetStickiness();
     } else if (
       !_.isEqual(prevProps.parameterValues, this.props.parameterValues) ||
-      (!prevProps.dashboard && this.props.dashboard)
+      (!prevProps.dashboard && this.props.dashboard) ||
+      prevProps.selectedTabId !== this.props.selectedTabId
```

Comment:
We need to re-fetch the dashcards when changing tabs to apply filters to the cards on the new tab.